lib/pure/memfiles.nim

```diff
@@ -245,6 +249,26 @@ proc open*(filename: string, mode: FileMode = fmRead,
       if close(result.handle) == 0:
         result.handle = -1
 
+proc flush*(f: var MemFile; attempts: uint = 3) =
+  ## Flushes `f`'s buffer for the number of attempts equal to `attempts`.
+  ## If were errors an exception `OSError` will be raised.
+  when defined(windows):
+    var res = false
+    for i in 1..attempts:
+      res = flushViewOfFile(f.mem, 0) != 0
```

Comment:
Again, what dom said, why not base the *retry* on the return value of ``flushViewOfFile``?

Reply:
Currently, an exception will be raised due to any OS error.
The developer has to decide what to do next.

Reply:
Ignoring some specific errors, it's a bad design.

Reply:
No, it's *retrying* only on specific errors, it's good design. Of course you should raise an exception for any other error.

Reply:
Well, then I need to investigate/experiments with flags and error codes of msync, too. :)